sdk/storage/Azure.Storage.DataMovement/src/ServiceToServiceTransferJob.cs

```diff
@@ -108,9 +108,9 @@ await foreach (JobPartInternal part in GetStorageResourcesAsync().ConfigureAwait
                 bool isFinalPartFound = false;
                 foreach (JobPartInternal part in _jobParts)
                 {
-                    if (part.JobPartStatus != DataTransferStatus.Completed)
+                    if (!part.JobPartStatus.HasCompletedSuccessfully)
                     {
-                        part.JobPartStatus = DataTransferStatus.Queued;
+                        part.JobPartStatus.OnTransferStateChange(DataTransferStatus.TransferState.Queued);
```

Comment:
Seeing this "OnXyz" naming get used makes me wonder if that's what we really want. The naming indicates to me that it's a side effect of the state changing. But this is the actual changing of the new state. We may just want to use more traditional "set" terminology. Or maybe "report."

Reply:
done. Changed it to `TrySet..`